engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobUpdateTimeoutTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import static io.camunda.zeebe.test.util.record.RecordingExporter.jobRecords;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import io.camunda.zeebe.test.util.Strings;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class JobUpdateTimeoutTest {
```

Comment:
:x: We should also add some tenant-related tests:
* Ensure that a job belonging to a "custom" tenant, can have it's deadline updated (with an authorization of the same tenant).
* Ensure that a job belonging to a "custom-A" tenant, **can't** have it's deadline updated by a "custom-B" tenant.

Similar tests are implemented [here](https://github.com/camunda/zeebe/blob/ff15090290739f5932aa17c145a1bad9d0664841/engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobUpdateRetriesTest.java#L163-L207).